src/braze-upload/lib/db.ts

```diff
@@ -0,0 +1,48 @@
+import {Pool, QueryConfig, QueryResult} from 'pg';
+import {logInfo} from "./log";
+
+export type DBConfig = {
+    username: string,
+    url: string,
+    password: string
+};
+
+export function createDatabaseConnectionPool(dbConfig: DBConfig): Pool {
+    const match = dbConfig.url.match(/\/\/(.*)\/(.*)/);
+    if (match !== null) {
+        const [_, host, database] = match;
+        return new Pool({
+            host,
+            database,
+            user: dbConfig.username,
+            password: dbConfig.password,
+
+            port: 5432,
+        });
+    }
+
+    throw new Error(`Could not parse DB config ${JSON.stringify(dbConfig)}`);
+}
+
+function queryToString(queryConfig: QueryConfig): string {
+    return `Query: ${JSON.stringify(queryConfig)}`;
+}
+
+export function fetchReferralData(referralCode: string, pool: Pool): Promise<QueryResult> {
```

Comment:
Much of this file is copied from the other `db.ts` (as building with shared code is proving difficult).
`fetchReferralData` is the novel part